plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java

```diff
@@ -170,6 +172,7 @@ private IcebergFileWriter createParquetWriter(
                     .setMaxPageValueCount(getParquetWriterPageValueCount(session))
                     .setMaxBlockSize(getParquetWriterBlockSize(session))
                     .setBatchSize(getParquetWriterBatchSize(session))
+                    .setBloomFilterColumns(getParquetBloomFilterColumns(storageProperties))
```

Comment:
a few lines above you have the _original_ `fileColumnNames` - please use those in correlation with what is specified in the table properties (case insensitive name matching) in `getParquetBloomFilterColumns`.

Also a new test to add: schema evolution - create a table with a bunch of bloom filter columns, drop one of the columns which was specified as bloom filter column and make sure that you don't get any errors . I'm guessing we'd have to filter out in `getParquetBloomFilterColumns` the column names which don't exist anymore.

Reply:
The writing logic ignores non-existent columns for which the Bloom filter property is set.